src/rust/aggregator/src/pipelines/leaderboards.rs

```diff
@@ -76,10 +76,7 @@ impl Pipeline for Leaderboards {
         for comp in competitions {
             aggregate_data_for_competition(&mut transaction, comp).await?;
         }
-        transaction
-            .commit()
-            .await
-            .map_err(|e| PipelineError::ProcessingError(anyhow!(e)))?;
+        commit_transaction(transaction).await?;
```

Comment:
If you're going to update this, how about also abstracting in the new util function to create repeatable read txn?